src/dichotomy.cc

```diff
@@ -167,18 +167,18 @@ void Encoder::LoopScan() {
   SetQuantMatrices(opt_quants);
   for (int c = 0; c < 2; ++c) FinalizeQuantMatrix(&quants_[c], q_bias_);
 
-  // return informative values to the user
-  search_hook_->q = best_q;
-  search_hook_->value = best_result;
-
   // optimize Huffman table now, if we haven't already during the search
-  if (!search_hook_->for_size) {
+  if (!search_hook_->for_size || best_q != search_hook_->q) {
```

Comment:
at line 126, we could declare:
  bool last_is_best = false;

and at line 154, we could have:

last_is_best = (p == 0 || fabs(result - search_hook_->target) < best);
if (last_is_best) {
 ...
}

Then this line would just be: if (!search_hook_->for_size || !last_is_best) {
   ...
}


WDYT?

Reply:
Yes, this is more explicit, thanks. Updated the PR.